### PR TITLE
[13.0] [MIG] website_sale_product_attribute_value_filter_existing: Migration to v13

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT24146

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Check Travis

**Comment by CarlosRoca13** [comment]
@pedrobaeza It is already solved

File: website_sale_product_attribute_value_filter_existing/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2019 Tecnativa - Victor M.M. Torres
```
Comment (yajo) [inline website_sale_product_attribute_value_filter_existing/__manifest__.py:1]:
```suggestion
# Copyright 2020 Tecnativa - Victor M.M. Torres
```

Same in all others.

File: website_sale_product_attribute_value_filter_existing/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+{
+    "name": "Website Sale Attribute Value Existing",
+    "summary": "Allow hide attributes values not used in variants",
+    "version": "13.0.1.0.0",
+    "development_status": "Stable",
+    "category": "Website",
+    "website": "https://github.com/OCA/e-commerce",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "license": "LGPL-3",
```
Comment (yajo) [inline website_sale_product_attribute_value_filter_existing/__manifest__.py:11]: Is this on purpose? @pedrobaeza

File: website_sale_product_attribute_value_filter_existing/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class ProductAttributeValues(WebsiteSale):
+    def _get_search_domain(
+        self, search, category, attrib_values, search_in_description=True
+    ):
+        # Store used domain in context to be reused after
+        domain = super(ProductAttributeValues, self)._get_search_domain(
```
Comment (yajo) [inline website_sale_product_attribute_value_filter_existing/controllers/main.py]: We're on python 3 now! :yum: 

```suggestion
        domain = super()._get_search_domain(
```

File: website_sale_product_attribute_value_filter_existing/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            search, category, attrib_values, search_in_description=search_in_description
+        )
+        new_context = dict(request.env.context, shop_search_domain=domain)
+        request.context = new_context
+        return domain
+
+    @http.route()
+    def shop(self, page=0, category=None, search="", ppg=False, **post):
+        res = super(ProductAttributeValues, self).shop(
```
Comment (yajo) [inline website_sale_product_attribute_value_filter_existing/controllers/main.py]:
```suggestion
        res = super().shop(
```

File: website_sale_product_attribute_value_filter_existing/i18n/es.po [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+#~ msgstr "Color especial de prueba"
+
+#~ msgid "Test blue"
+#~ msgstr "azul"
+
+#~ msgid "Test green"
+#~ msgstr "verde"
+
+#~ msgid "Test red"
+#~ msgstr "rojo"
```
Comment (yajo) [inline website_sale_product_attribute_value_filter_existing/i18n/es.po]: Update this file or delete it please.

File: website_sale_product_attribute_value_filter_existing/data/demo.xml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
```
Comment (yajo) [inline website_sale_product_attribute_value_filter_existing/data/demo.xml]: :no_entry_sign: This file is not declared as demo in the manifest, so either add it to the manifest or remove it from here.

Also, if you add it, it should be named something like `demo/product_attribute.xml`

File: website_sale_product_attribute_value_filter_existing/views/templates.xml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2019 Tecnativa - Victor M.M. Torres
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). -->
+<odoo>
+    <template id="products_attributes" inherit_id="website_sale.products_attributes">
+        <xpath
+            expr="//t[@t-if=&quot;a.display_type == &apos;select&apos;&quot;]//t[@t-foreach='a.value_ids']/option"
```
Comment (yajo) [inline website_sale_product_attribute_value_filter_existing/views/templates.xml]: No need to escape these:

```suggestion
            expr="//t[@t-if=&quot;a.display_type == 'select'&quot;]//t[@t-foreach='a.value_ids']/option"
```

File: website_sale_product_attribute_value_filter_existing/views/templates.xml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+            position="attributes"
+        >
+            <attribute name="t-if">attr_values_used &amp; v</attribute>
+        </xpath>
+        <xpath
+            expr="//t[@t-if=&quot;a.display_type == &apos;radio&apos;&quot;]//t[@t-foreach='a.value_ids']/li"
```
Comment (yajo) [inline website_sale_product_attribute_value_filter_existing/views/templates.xml]:
```suggestion
            expr="//t[@t-if=&quot;a.display_type == 'radio'&quot;]//t[@t-foreach='a.value_ids']/li"
```

File: website_sale_product_attribute_value_filter_existing/views/templates.xml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+            position="attributes"
+        >
+            <attribute name="t-if">attr_values_used &amp; v</attribute>
+        </xpath>
+        <xpath
+            expr="//t[@t-if=&quot;a.display_type == &apos;color&apos;&quot;]//t[@t-foreach='a.value_ids']/label"
```
Comment (yajo) [inline website_sale_product_attribute_value_filter_existing/views/templates.xml]:
```suggestion
            expr="//t[@t-if=&quot;a.display_type == 'color'&quot;]//t[@t-foreach='a.value_ids']/label"
```

**Review by yajo — CHANGES_REQUESTED** [review]
Just one important comment; the rest are just comments or suggestions.

File: website_sale_product_attribute_value_filter_existing/__manifest__.py (hunk 1 above)
Comment (pedrobaeza) [reply website_sale_product_attribute_value_filter_existing/__manifest__.py:1]: No, you mustn't change the year for others copyright, as they haven't touched the files in this year.

File: website_sale_product_attribute_value_filter_existing/data/demo.xml (hunk 6 above)
Comment (pedrobaeza) [reply website_sale_product_attribute_value_filter_existing/data/demo.xml]: Better `demo/product_attribute_demo.xml`

File: website_sale_product_attribute_value_filter_existing/__manifest__.py (hunk 2 above)
Comment (pedrobaeza) [reply website_sale_product_attribute_value_filter_existing/__manifest__.py:11]: Yes, remember that we release all the website modules as LGPL.

**Comment by CarlosRoca13** [comment]
@Yajo all changes done!

File: website_sale_product_attribute_value_filter_existing/tests/test_website_sale_product_attribute_value_filter_existing.py [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        # Active attribute's filter in /shop. By default it's disabled.
+        self.env.ref("website_sale.products_attributes").active = True
+
+    def test_ui_website(self):
+        """Test frontend tour."""
+        tour = (
+            "odoo.__DEBUG__.services['web_tour.tour']",
+            "website_sale_product_attribute_value_filter_existing",
+        )
+        self.phantom_js(
```
Comment (sergio-teruel) [inline website_sale_product_attribute_value_filter_existing/tests/test_website_sale_product_attribute_value_filter_existing.py]: Change by self.browser_js

**Review by sergio-teruel — APPROVED** [review]
Only a little comment

**Review by HaraldPanten — APPROVED** [review]

**Review by yajo — APPROVED** [review]

File: website_sale_product_attribute_value_filter_existing/tests/test_website_sale_product_attribute_value_filter_existing.py (hunk 10 above)
Comment (pedrobaeza) [reply website_sale_product_attribute_value_filter_existing/tests/test_website_sale_product_attribute_value_filter_existing.py]: @CarlosRoca13 are you going to change this before merging?

Comment (CarlosRoca13) [reply website_sale_product_attribute_value_filter_existing/tests/test_website_sale_product_attribute_value_filter_existing.py]: Change is already done, sorry I didn't saw the comment. 

Thanks for the reminder

**Pull request merged**